### PR TITLE
Document Checkpoints

### PR DESCRIPTION
**Pull request by jmjatlanta**
Provides better documentation of checkpoints. No change in functionality

### CODE REVIEW

**Review by DeckerSU — COMMENTED** [review]
Could you plz keep the original includes order and forward declarations in this PR. Leave only documentation additions. Also i would like to keep the following constructions:
```
#ifndef BITCOIN_CHECKPOINTS_H
#define BITCOIN_CHECKPOINTS_H
...
#endif // BITCOIN_CHECKPOINTS_H
```
Instead of just:
```
#pragma once
```
For the new source files i don't mind of `#pragma once` usage, but for sources inherited from Bitcoin, better to keep "existing order" of things. Just bcz it will help/promote to easy cherry-picking or merge some PRs from bitcoin upstream.

In other words i would like to keep document PRs - "document only". Such type of PRs should only add comments and don't touch includes, defines, pragmas, etc. IMHO.

**Comment by dimxy** [comment]
tbh i'd prefer to keep original files intact, if possible

**Comment by tonymorony** [comment]
these changes implemented in combined PR: https://github.com/KomodoPlatform/komodo/pull/559
